src/js/lectureRoom/displayVideos.js

```diff
@@ -0,0 +1,122 @@
+  renderDependsOnTabState,
+} from 'utils/render';
+import fetchVideoId from 'utils/fetchVideoId';
+import globalState from 'utils/globalState';
+
+const $mainVideoWrapper = document.querySelector('main .video-wrapper');
+const $nav = document.querySelector('nav');
+
+let stateChangedItems = [];
+let notDeletedItems = [];
+
+const onClickWatched = e => {
+  const { target } = e;
+
+  if (!target.matches('.watched')) return;
+
+  const savedItems = JSON.parse(localStorage.getItem('savedItems'));
+  const clickedVideoId = fetchVideoId(target);
+
+  stateChangedItems = savedItems.map(savedItem => {
+    const {
+      id: { videoId },
+      state: { isWatched, isLiked },
+    } = savedItem;
+
+    return videoId === clickedVideoId
+      ? { ...savedItem, state: { isWatched: !isWatched, isLiked } }
+      : savedItem;
+  });
+
+  localStorage.setItem('savedItems', JSON.stringify(stateChangedItems));
+
+  renderDependsOnTabState(stateChangedItems, $mainVideoWrapper, renderSnackBar);
+};
+
+const onClickDelete = e => {
+  const { target } = e;
+
+  if (
+    !target.matches('.delete') ||
+    !window.confirm('이 영상을 삭제하시겠습니까?')
+  )
+    return;
+
+  const savedItems = JSON.parse(localStorage.getItem('savedItems'));
+  const clickedVideoId = fetchVideoId(target);
+
+  notDeletedItems = savedItems.filter(savedItem => {
+    const {
+      id: { videoId },
+    } = savedItem;
+
+    return videoId !== clickedVideoId;
+  });
+
+  localStorage.setItem('savedItems', JSON.stringify(notDeletedItems));
+
+  renderDependsOnTabState(notDeletedItems, $mainVideoWrapper);
+
+  renderSnackBar('해당 영상이 삭제되었습니다.');
+};
+
+const onClickLiked = e => {
+  const { target } = e;
+  if (!target.matches('.liked')) return;
+
+  const savedItems = JSON.parse(localStorage.getItem('savedItems'));
+  const clickedVideoId = fetchVideoId(target);
+
+  stateChangedItems = savedItems.map(savedItem => {
+    const {
+      id: { videoId },
+      state: { isWatched, isLiked },
+    } = savedItem;
+
+    return videoId === clickedVideoId
+      ? { ...savedItem, state: { isWatched, isLiked: !isLiked } }
+      : savedItem;
+  });
+
+  localStorage.setItem('savedItems', JSON.stringify(stateChangedItems));
+
+  renderDependsOnTabState(stateChangedItems, $mainVideoWrapper);
+};
+
+const onMenuChange = e => {
+  const { target } = e;
+
+  if (!target.matches('button')) return;
+
+  const savedItems = JSON.parse(localStorage.getItem('savedItems'));
+  const buttons = [...e.currentTarget.children];
+
+  if (target.id === 'notWatched') {
+    refreshItems($mainVideoWrapper);
+    renderNotWatchedItems(savedItems, $mainVideoWrapper);
+    globalState.tabState = 'notWatched';
+  } else if (target.id === 'watched') {
+    refreshItems($mainVideoWrapper);
+    renderWatchedItems(savedItems, $mainVideoWrapper);
+    globalState.tabState = 'watched';
+  } else {
+    refreshItems($mainVideoWrapper);
+    renderLikedItems(savedItems, $mainVideoWrapper);
+    globalState.tabState = 'liked';
+  }
+
```

Comment:
저도 원래 if 문이나 swtich문을 즐겨썼는데요~ 아래와 같이 객체로 작성해서  if 블락을 안도는 방법도 있는데 어떠신가용ㅎㅎㅎ이러면 더 빨라진다고해요 !
```suggestion
  const actions  = {
  notWatched : ()=>{ 
  },
  watched:()=>{
  },
  liked :() =>{
  }
  
  } 
  actions[globalState.tabState] && actions[globalState.tabState]();

```

Reply:
오우 이런 꿀팁을... 108번 라인은 옵셔널 체이닝 사용해서
`actions.?[globalState.tabState].?()` 이렇게 또는 어차피 actions라는 변수는 처음부터 객체로 초기화 되어있으니 프로퍼티 참조에서는 옵셔널 체이닝을 빼고 `actions[globalState.tabState].?()`
이렇게도 써줄 수 있을것 같아요ㅎㅎ

Reply:
오오.....이런 방법이...... 안그래도 친구가 이 방법을 썼다고 해서 인지만하고 있었지 실제로 적용해볼 생각을 못했네요.....
이제부턴 적극 활용을 해봐야겠네요!